File: dcDLLforUnity/dcDLLforUnity/dcAction.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO.Ports;
using System.Timers;

namespace dcDLLforUnity
{
    public class dcAction
    {
        public string arduinoPort = "COM4";
        public int baudRate = 115200;
        public SerialPort arduino;

        public dcAction(string port)
        {
            arduinoPort = port;
            baudRate = 115200;
        }
        public dcAction(string port, int baud)
        {
            arduinoPort = port;
            baudRate = baud;
        }

        public void Open()
        {
            arduino = new SerialPort(arduinoPort, baudRate);
            arduino.Open();
        }

        public void Write(string messages)
        {
            arduino.Write(messages);
        }

        public void WritePWM(int pwmValues)
        {
            string pwmString;
            if (pwmValues < 100 && pwmValues >= 10) { pwmString = "0" + pwmValues.ToString(); }
            else if (pwmValues < 10 && pwmValues >= 0) { pwmString = "00" + pwmValues.ToString(); }
            else { pwmString = pwmValues.ToString(); }
            arduino.Write("M" + pwmString);
        }

        public void Close()
        {
            WritePWM(0); // make sure to set PWM = 0
            arduino.Close();
        }

        public bool IsOpen()
        {
            return arduino.IsOpen;
        }

        public string ReadLine()
        {
            return arduino.ReadLine();
        }

        public string ReadTo(string values)
        {
            return arduino.ReadTo(values);
        }
    }
}

File: dcPythonLib/dcAction.py
import serial
import threading


class dcAction(threading.Thread):
    def __init__(self, port="COM4", baudRate=115200):
        threading.Thread.__init__(self)
        # self.port = port
        # self.baudRate = baudRate
        self.arduino = serial.Serial(port=port, baudrate=baudRate)

    def Open(self):
        # arduino = serial.Serial(port=self.port, baudrate=self.baudRate)
        # arduino.open()
        self.arduino.open()

    def Write(self, messages):
        self.arduino.write(messages)

    def WritePWM(self, pwmValues):
        if (pwmValues < 100 and pwmValues >= 10):
            pwmString = "0" + str(pwmValues)
        elif (pwmValues < 10 and pwmValues >= 0):
            pwmString = "00" + str(pwmValues)
        else:
            pwmString = str(pwmValues)

        self.arduino.write("M" + pwmString)

    def Close(self):
        self.WritePWM(0)
        self.arduino.close()

    def IsOpen(self):
        return self.arduino._isOpen

    def ReadLine(self):
        return self.arduino.readline()

# # connect to arduino on mac
# macDuino = dcAction(port="/dev/cu.usbmodem1411")
# # open arduino connection
# # macDuino.Open()
# driveSpeed = 0
#
# while (driveSpeed != 256):
#     driveSpeed = input("speed you want?: ")
#     macDuino.WritePWM(driveSpeed)
#
# macDuino.Close()
